src/olympia/files/tests/test_models.py

```diff
@@ -216,6 +216,17 @@ def test_generate_filename_ja(self):
             == '42/4242/4242/addon-0.1.7.xpi'
         )
 
+    def test_generate_filename_no_slug(self):
+        file_ = File()
+        file_.version = Version(version='0.1.7')
+        file_.version.compatible_apps = {amo.FIREFOX: None}
+        file_.version.addon = Addon(slug='', pk=4242)
```

Comment:
can we test when `slug=None` also?

Reply:
https://github.com/mozilla/addons-server/pull/19266/commits/08c1b191997b6e6c75c31f3bdd3755187e6f05a3